### PR TITLE
Remove unneeded `conan install` commands.

### PR DESCRIPTION
**Pull request by cqc-alec**
I learned that in conan 2, `conan build` automatically does `conan install` first. Therefore we can delete these `conan install` commands (as long as we add `--build=missing` where needed to the subsequent `conan build` commands).

### CODE REVIEW

File: .github/workflows/test_libs.yml [hunk 1]
```diff
@@ -133,13 +133,11 @@ jobs:
...
-        conan build libs/${{ matrix.lib }} -s build_type=Debug -o boost/*:header_only=True -o ${{ matrix.lib }}/*:profile_coverage=True -of build/${{ matrix.lib }}
+        conan build libs/${{ matrix.lib }} -s build_type=Debug --build=missing -o boost/*:header_only=True -o ${{ matrix.lib }}/*:profile_coverage=True -of build/${{ matrix.lib }}
         conan export-pkg libs/${{ matrix.lib }} -s build_type=Debug -o boost/*:header_only=True -o ${{ matrix.lib }}/*:profile_coverage=True -of build/${{ matrix.lib }} -tf ""
     - name: build ${{ matrix.lib }} tests
       run: |
-        conan install libs/${{ matrix.lib }}/test -s build_type=Debug --build=missing -o boost/*:header_only=True -o test-${{ matrix.lib }}/*:with_coverage=True -of build/${{ matrix.lib }}-tests
-        conan build libs/${{ matrix.lib }}/test -s build_type=Debug --build=missing -o boost/*:header_only=True -o test-${{ matrix.lib }}/*:with_coverage=True -of build/${{ matrix.lib }}-tests
+        conan build   libs/${{ matrix.lib }}/test -s build_type=Debug --build=missing -o boost/*:header_only=True -o test-${{ matrix.lib }}/*:with_coverage=True -of build/${{ matrix.lib }}-tests
```
Comment (doug-q) [inline .github/workflows/test_libs.yml:140]:
```suggestion
        conan build libs/${{ matrix.lib }}/test -s build_type=Debug --build=missing -o boost/*:header_only=True -o test-${{ matrix.lib }}/*:with_coverage=True -of build/${{ matrix.lib }}-tests
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -192,14 +192,12 @@ jobs:
         # On windows use install + build so build directory is consistent across runs (i.e. don't build from conan cache)
         # Necessary because setting ccache base_dir doesn't currently work on windows
```
Comment (doug-q) [inline .github/workflows/build_and_test.yml:193]: This comment is now out of date. Can it go, or does it mean we shouldn't remove the install on windows?

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -192,14 +192,12 @@ jobs:
...
         run: |
-          conan install tket --user=tket --channel=stable -o boost/*:header_only=True -o with_all_tests=True -c tools.cmake.cmaketoolchain:generator=Ninja
           conan build tket --user=tket --channel=stable -o boost/*:header_only=True -o with_all_tests=True -c tools.cmake.cmaketoolchain:generator=Ninja
```
Comment (doug-q) [inline .github/workflows/build_and_test.yml:195]: Do we need `--build=missing` here? (and on line 202)

Comment (cqc-alec) [reply .github/workflows/build_and_test.yml:195]: No, we only need to add it when it was included in the preceding "install" command being removed.

**Review by doug-q — APPROVED** [review]
:+1:

**Pull request merged**